byterun/array.c

```diff
@@ -42,39 +45,72 @@ CAMLexport int caml_is_double_array(value array)
   return (Tag_val(array) == Double_array_tag);
 }
 
+/* [ 'a array -> int -> 'a ] where 'a != float */
 CAMLprim value caml_array_get_addr(value array, value index)
 {
   intnat idx = Long_val(index);
   if (idx < 0 || idx >= Wosize_val(array)) caml_array_bound_error();
   return Field(array, idx);
 }
 
+/* [ float array -> int -> float ] */
 CAMLprim value caml_array_get_float(value array, value index)
 {
   intnat idx = Long_val(index);
+#ifdef FLAT_FLOAT_ARRAY
   double d;
   value res;
 
   if (idx < 0 || idx >= Wosize_val(array) / Double_wosize)
     caml_array_bound_error();
-  d = Double_field(array, idx);
+  d = Double_flat_field(array, idx);
 #define Setup_for_gc
 #define Restore_after_gc
   Alloc_small(res, Double_wosize, Double_tag);
 #undef Setup_for_gc
 #undef Restore_after_gc
   Store_double_val(res, d);
   return res;
+#else /* FLAT_FLOAT_ARRAY */
```

Comment:
This comment is confusing -- it looks like the "else" branch is when the float array hack is enabled, which it's not.

Reply:
(This has been addressed)